atlasdb-commons/src/main/java/com/palantir/common/time/NanoTime.java

```diff
@@ -14,19 +14,25 @@
  * limitations under the License.
  */
 
-package com.palantir.leader.lease;
+package com.palantir.common.time;
 
 import java.time.Duration;
 import java.util.Objects;
 import java.util.concurrent.locks.LockSupport;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import com.google.common.annotations.VisibleForTesting;
 
 public final class NanoTime implements Comparable<NanoTime> {
+    @JsonProperty("time")
     private final long time;
 
-    @VisibleForTesting
-    NanoTime(long time) {
+    @JsonCreator
+    public NanoTime(@JsonProperty("time") long time) {
```

Comment:
I would omit the `@JsonProperty("time")` and just always serialize a nanotime as a long?

Reply:
I somehow get the impression that prod code should only be creating this by calling NanoTime.now(). How about letting this be called only via a method `createForTests`?